lib/solvers/SchematicTracePipelineSolver/SchematicTracePipelineSolver.ts

```diff
@@ -134,6 +136,34 @@ export class SchematicTracePipelineSolver extends BaseSolver {
         },
       },
     ),
+    definePipelineStep(
+      "traceLabelOverlapAvoidanceSolver",
+      TraceLabelOverlapAvoidanceSolver,
+      (instance) => {
+        const traceMap =
+          instance.traceOverlapShiftSolver?.correctedTraceMap ??
```

Comment:
make sure this isn't being mutated, based on the changes i think it's being mutated. We don't want solvers to mutate their inputs because it breaks input -> output pipelining

introduce a `getOutputTraceMap()` or some similar function that can be used by `visualize()`, tscircuit/core will need to be adapted to use this function because i believe it currently goes into the solver and uses the `correctedTraceMap` (which is bad, we should have always had `getOutputTraceMap()` so that we could extend the pipeline easily